app/models/optin/OptInContextData.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.optin
+
+import models.itsaStatus.ITSAStatus
+import models.itsaStatus.ITSAStatus.ITSAStatus
+import play.api.libs.json.{Json, OFormat}
+
+case class OptInContextData(crystallisationStatus: Boolean,
+                             previousYearITSAStatus: String,
+                             currentYearITSAStatus: String,
+                             nextYearITSAStatus: String)
+
+object OptInContextData {
+  implicit val format: OFormat[OptInContextData] = Json.format[OptInContextData]
+
+  def statusToString(status: ITSAStatus): String =
```

Comment:
This is all duplicated from Opt Out. We should look to unify this.

Reply:
I'll do this in the service pr

Reply:
Cool